### PR TITLE
context management switch

### PR DESCRIPTION
**Pull request by 0xToshii**
### Related Issue

**Issue:** #XXXX

### Description

This PR adds a menu item which allows you to trigger using the new or old context management methods. There will be a fast follow to this where this triggers the new context management method only for next gen models, as that will also require doc changes covering this.

The old context management method has been brought back, and there is now effectively a state var which triggers using one method or the other. Both methods use the underlying context management disk system, so there should be no issues in them being triggered one after the other, if the user decides to use both in tandem.

We use clear branching in `task/index.ts::recursivelyMakeClineRequests` for this even through there is slight code re-use because it's much cleaner.

### Test Procedure

Test that selecting the menu item actually changes which method is used (easiest way is to add logging). There are some valid edge cases to test as well including what if we cancel the summarization request from the new context managment tool call, select using the old context management method in the menu, then doing another request.

### Type of Change

-   [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
-   [x] ✨ New feature (non-breaking change which adds functionality)
-   [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] ♻️ Refactor Changes
-   [ ] 💅 Cosmetic Changes
-   [ ] 📚 Documentation update
-   [ ] 🏃 Workflow Changes

### Pre-flight Checklist

-   [x] Changes are limited to a single feature, bugfix or chore (split larger changes into separate PRs)
-   [x] Tests are passing (`npm test`) and code is formatted and linted (`npm run format && npm run lint`)
-   [ ] I have created a changeset using `npm run changeset` (required for user-facing changes)
-   [ ] I have reviewed [contributor guidelines](https://github.com/cline/cline/blob/main/CONTRIBUTING.md)

### Screenshots

Screenshot 2025-08-19 at 2 51 55 PM

### Additional Notes

----

> [!IMPORTANT]
> Introduces an auto-condense feature for context management, with backend logic and UI integration for user control.
> 
>   - **Behavior**:
>     - Introduces `useAutoCondense` option in `UpdateSettingsRequest` in `state.proto`.
>     - Adds `useAutoCondense` parameter to `getNewContextMessagesAndMetadata()` in `ContextManager.ts`.
>     - Implements logic for auto-condense in `getNewContextMessagesAndMetadata()` and `recursivelyMakeClineRequests()` in `Task.ts`.
>     - Updates `initTask()` and `updateSettings()` in `Controller` to handle `useAutoCondense`.
>   - **State Management**:
>     - Adds `useAutoCondense` to `GlobalState` and `GlobalStateKey` in `state-keys.ts`.
>     - Updates `CacheService` to manage `useAutoCondense` state.
>     - Modifies `readStateFromDisk()` in `state-helpers.ts` to include `useAutoCondense`.
>   - **UI**:
>     - Adds "Enable auto condense" checkbox to `FeatureSettingsSection.tsx`.
>     - Updates `ExtensionStateContext.tsx` to manage `useAutoCondense` state.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=cline%2Fcline&utm_source=github&utm_medium=referral)<sup> for 9a0f5a5251982a534ec6844f0cd35431afd88c95. You can [customize](https://app.ellipsis.dev/cline/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: webview-ui/src/components/settings/sections/FeatureSettingsSection.tsx [hunk 1]
```diff
@@ -167,6 +168,20 @@ const FeatureSettingsSection = ({ renderSectionHeader }: FeatureSettingsSectionP
...
+						<VSCodeCheckbox
+							checked={useAutoCondense}
+							onChange={(e: any) => {
+								const checked = e.target.checked === true
+								updateSetting("useAutoCondense", checked)
+							}}>
+							Enable auto condense
+						</VSCodeCheckbox>
+						<p className="text-xs text-[var(--vscode-descriptionForeground)]">
+							Enables advanced context management system which uses LLM based condensing, instead of rule-based
```
Comment (pashpashpash) [inline webview-ui/src/components/settings/sections/FeatureSettingsSection.tsx:181]: Let's update the docs, and mention that this is only enabled for the latest frontier models from xAI, OpenAI, Anthropic, and Google, and link to the docs.

**Review by pashpashpash — CHANGES_REQUESTED** [review]

File: webview-ui/src/components/settings/sections/FeatureSettingsSection.tsx (hunk 1 above)
Comment (0xToshii) [reply webview-ui/src/components/settings/sections/FeatureSettingsSection.tsx:181]: im adding this in the next pr

**Comment by 0xToshii** [comment]
merged upstream
